backend/parser/listener/listener.go

```diff
@@ -300,6 +300,10 @@ func (s *searchListener[T]) appendRules(value string) {
 		filterKey = fmt.Sprintf("toString(%s)", filterKey)
 	}
 
+	// Quotes are sometimes escaped on the client and need to be unescaped before
+	// being used in the query or they will be double escaped.
+	value = strings.ReplaceAll(value, `\"`, `"`)
```

Comment:
@mayberryzane curious if you can think of a better way to handle this, and quote escaping in general on queries 🤔